**AMR**

1. *Resfinder*

[Source Page](https://cge.cbs.dtu.dk/services/ResFinder/)

Just upload fasta file
Choose the perimeters 

2. *RGI - Resistance gene identifier*

[Source Page](https://card.mcmaster.ca/analyze/rgi) 

Just add fasta file

3. *Abricate* 

[Source Page](https://github.com/tseemann/abricate)

[Tutorial](https://www.youtube.com/watch?v=2SKrbweUFr8)  - 15.19 min

Code-

`abricate *.fasta` - for multiple files

`abricate Your_file.fasta`  - for single file

Make table for resistance to merge data

**Check Criteria for MDR, XDR, PDR**

[Source Page](https://www.clinicalmicrobiologyandinfection.com/article/S1198-743X(14)61632-3/fulltext) Check this Article for details.


4. Use *Kleborate*

[Source Page and Details]https://github.com/klebgenomics/Kleborate

I havent used Kleborate yet but this is the go to for Klebsiella spp. 
