**Phylogeny Tree**

[Tutorial](https://www.youtube.com/watch?v=7GAYLbiyLuw)

Steps 

1. To construct the initial phylogenetic tree, generate the 16S RNA sequence using Barrnap. 

2. Then, Blastn to find and identify the most similar matches
[Tutorial](https://blast.ncbi.nlm.nih.gov/Blast.cgi) 

3. The software MEGA11 needs to be installed 

[Download page]https://www.megasoftware.net/

4. Make a new Text file and copy the sequences of your and similar matches sequences following the rules - 

>Name_of_organism_Location_Year  Change this according but *>* is mandatory

5. Go to Mega and click *ALign* and Build new Alignnment and choose New Alignment.

6. Choose DNA and then Input your sequence file by pressing *Ctrl+D* 

7. Check for mistakes and then use the muscle algorithm to alighn them. Choose UPGMA option to construct a rooted phylogenetic tree. Save the file in Mega format.

8. Go back to MEGA application and choose Phylogeny and choose the tree type you want and keep default parameters. Just keep p-value if its not chosen.

9. Save the tree as you like.
